updated doc with troubleshooting section for console logout issue (#2900)
kubeapps/kubeapps: docs/user/using-an-OIDC-provider.md

### Checking the logs of your Kubernetes API server

Finally, if none of the above are relevant to your issue, you can check the logs of the Kubernetes API server deployment for OIDC-related lines at the time of your login attempt. These may show a configuration issue with the API server itself.

### User automatically logged out from Kubeapps Console

When using the default auth proxy, some users may experience the behavior where they are automatically logged out from the console.

> The default auth proxy is not configured to refresh the access/openid token and the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5mn.

Reviewer: Isn't this statement going to be incorrect once your chart change lands (as it's setting a sensible default, which ever way it does so)?

Author: this will be obsolete for new installation / upgrades once we have the new chart.
but this is an existing problem for users, so i think it is useful to have a troubleshooting section for it.
I agree that i will have to reword it once we have the change in the chart.

> To avoid this issue, add the option `--cookie-refresh=2m` to `authProxy.additionalFlags` in your values file. The duration for the refresh must be lesser than the access/openid expiration time configured in the OAuth2/OIDC provider.

Reviewer: Similar to my comment on the chart, I think this would be a better UX to be able to just set a chart value directly. Guess we'll want to wait until the chart change is landed before updating this.

Author: yes, this was worded with assumption the new chart is not yet available.
if we wait for the chart, i will reword accordingly (reason why i marked the PR as draft :-)

### Checking the logs of your Kubernetes API server

Finally, if none of the above are relevant to your issue, you can check the logs of the Kubernetes API server deployment for OIDC-related lines at the time of your login attempt. These may show a configuration issue with the API server itself.

### User automatically logged out from Kubeapps Console

When using the default auth proxy, some users may experience the behavior where they are automatically logged out from the console.

> The default auth proxy (prior to version 7.0.5) is not configured to refresh the access/openid token and the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5mn.

Reviewer: ```suggestion
Prior to Kubeapps 7.0.5, the auth proxy configuration did not include a default `--cookie-refresh` value to refresh the access/openid token and so the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5m.
```

> Prior to Kubeapps 7.0.5, the auth proxy configuration did not include a default `--cookie-refresh` value to refresh the access/openid token and so the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5m.

Reviewer 2: ```suggestion
Prior to the Kubeapps chart version 7.1.0, the auth proxy configuration did not include a default `--cookie-refresh` value to refresh the access/openid token and so the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5m.
```

To avoid this issue, you can do one of the following:

>  - upgrade Kubeapps to version 7.0.5+

Reviewer: ```suggestion
 - upgrade Kubeapps to version 7.0.5+ which sets a default of `--cookie-refresh=2m` and exposes the value in the chart values as `authProxy.cookieRefresh`.
```

>  - upgrade Kubeapps to version 7.0.5+ which sets a default of `--cookie-refresh=2m` and exposes the value in the chart values as `authProxy.cookieRefresh`.

Reviewer 2: ```suggestion
 - upgrade the Kubeapps chart to version 7.1.0+ which sets a default of `--cookie-refresh=2m` and exposes the value in the chart values as `authProxy.cookieRefresh`.
```

 - update Kubeapps by adding the option `--cookie-refresh=2m` to `authProxy.additionalFlags`.

The duration for the refresh must be lesser than the access/openid expiration time configured in the OAuth2/OIDC provider.

> **Note**: The issue may still occur even after upgrading or updating Kubeapps as OAuth2 Proxy does not support cookie-refresh for all providers. See [OAuth2 Proxy Configuration Overview](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview/#footnote1) for the list of supported providers.

Reviewer: Not sure how relevant this is given that we recommend the `oidc` provider which is supported. But yep, people aren't restricted to that (eg. I've used the google provider in certain cases). Perhaps:
```suggestion
**Note**: If you have configured a provider other than `oidc` for oauth2-proxy, the issue may still occur even after upgrading or updating Kubeapps as OAuth2 Proxy does not support cookie-refresh for all providers. See [OAuth2 Proxy Configuration Overview](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview/#footnote1) for the list of supported providers.
```
